pkg/automationconfig/automation_config_builder.go

```diff
@@ -61,6 +64,17 @@ func (b *Builder) SetFCV(fcv string) *Builder {
 	return b
 }
 
+func (b *Builder) SetTLS(caFile, certAndKeyFile string, mode SSLMode) *Builder {
+	b.tlsCAFile = caFile
+	b.tlsCertAndKeyFile = certAndKeyFile
+	b.tlsMode = mode
+	return b
+}
+
+func (b *Builder) isTLSEnabled() bool {
+	return b.tlsCAFile != "" && b.tlsCertAndKeyFile != ""
```

Comment:
one thing to watch out for, it's a valid configuration for these fields to be non-empty, if mode is set to `disabled` , in this case this function would incorrectly return `true`

Reply:
Good catch, fixed!